packages/engine/src/datastore/table/state/mod.rs

```diff
@@ -38,23 +38,35 @@ pub struct Inner {
     local_meta: Meta,
 
     num_agents: usize,
+
+    sim_config: Arc<SimRunConfig>,
 }
 
-impl Inner {
-    /// Creates a new State object from a collection of groups of `AgentState`.
+impl State {
+    /// Creates a new State object from a provided array of `AgentState`.
     ///
     /// Uses the schemas in the provided `sim_config` to validate the provided state, and to create
-    /// the agent and message batches. The agent batches use the data provided in
-    /// `agent_state_batches`, where each element is a group, and the total elements (i.e.
-    /// `AgentState`s) within those groups is `num_agents`.
-    ///
-    /// Effectively converts the `agent_state_batches` from Array-of-Structs into a
-    /// Struct-of-Arrays.
-    fn from_agent_states(
-        agent_state_batches: &[&[AgentState]],
-        num_agents: usize,
-        sim_config: &SimRunConfig,
-    ) -> Result<Inner> {
+    /// the agent and message batches. The agent batches are created by splitting up the
+    /// `agent_states` into groups based on the number of workers specified in `sim_config`.
+    pub fn from_agent_states(
+        agent_states: &[AgentState],
+        sim_config: Arc<SimRunConfig>,
+    ) -> Result<State> {
+        let num_workers = sim_config.sim.engine.num_workers;
+        let num_agents = agent_states.len();
+
+        // Distribute agents across workers
+        const MIN_PER_WORKER: usize = 10;
+        let num_agents_per_worker =
+            ((num_agents as f64 / num_workers as f64).ceil() as usize).max(MIN_PER_WORKER);
+        let mut agent_state_batches = vec![];
```

Comment:
I would rename this to `agent_state_groups`, since they're `Vec`s rather than Arrow batches and also to clarify that each batch belongs to a separate group